File: Interview Preparation Kit/Array/Arrays Ds/Arrays-DS.cpp
#include <vector>
#include <algorithm> // For std::swap

/*
    Time Complexity: O(n)
    Space Complexity: O(1)
*/
std::vector<int> reverseArray(std::vector<int> a) {
    int l = 0;                   // Left pointer at the start of the array
    int r = a.size() - 1;         // Right pointer at the end of the array

    // Swap elements from both ends towards the middle
    while (l < r) {               
        std::swap(a[l], a[r]);    // Swap left and right elements
        l++;                      // Move left pointer forward
        r--;                      // Move right pointer backward
    }

    return a; // Return the reversed array
}

File: Interview Preparation Kit/Array/Arrays Ds/Arrays-DS.py
# Time Complexity: O(n)
# Space Complexty: O(1)
def reverse_array(arr):
    l, r = 0, len(arr) - 1  # Initialize left and right pointers
    
    # Swap elements from both ends towards the middle
    while l < r:
        arr[l], arr[r] = arr[r], arr[l]  # Swap elements
        l += 1  # Move left pointer forward
        r -= 1  # Move right pointer backward
    
    return arr  # Return the reversed array
